packages/wallet/src.ts/index.ts

```diff
@@ -220,6 +289,68 @@ export class Wallet extends Signer implements ExternallyOwnedAccount, TypedDataS
 		return encryptKeystore(this, password, options, progressCallback);
 	}
 
+	async call(txRequest : Deferrable<TransactionRequest>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+		this._checkProvider("call");
+		const tx = await resolveProperties(this.checkTransaction(txRequest));
+		const contractAccountLikeID = getAccountFromAddress(tx.to.toString());
+		const contractId = `${contractAccountLikeID.shard}.${contractAccountLikeID.realm}.${contractAccountLikeID.num}`;
+
+		const thisAcc = getAccountFromAddress(await this.getAddress());
+		const thisAccId = `${thisAcc.shard}.${thisAcc.realm}.${thisAcc.num}`;
+
+		const nodeID = AccountId.fromString(tx.nodeId.toString());
+		const paymentTxId = TransactionId.generate(thisAccId);
+
+		const hederaTx = new ContractCallQuery()
+			.setContractId(contractId)
+			.setFunctionParameters(arrayify(tx.data))
+			.setNodeAccountIds([nodeID])
+			.setGas(Long.fromString(tx.gasLimit.toString()))
+			.setPaymentTransactionId(paymentTxId);
+
+		const paymentBody = {
+			transactionID: paymentTxId._toProtobuf(),
+			nodeAccountID: nodeID._toProtobuf(),
+			transactionFee: new Hbar(1).toTinybars(),
+			transactionValidDuration: {
+				seconds: Long.fromInt(120),
+			},
+			cryptoTransfer: {
+				transfers: {
+					accountAmounts:[
+						{
+							accountID: AccountId.fromString(thisAccId)._toProtobuf(),
+							amount: new Hbar(3).negated().toTinybars()
+						},
+						{
+							accountID: nodeID._toProtobuf(),
+							amount: new Hbar(3).toTinybars()
+						}
+					],
+				},
+			},
+		};
+
+		const signed = {
+			bodyBytes: TransactionBody.encode(paymentBody).finish(),
+			sigMap: {}
+		};
+
+		const walletKey = PrivateKey.fromStringECDSA(this._signingKey().privateKey);
+		const signature = walletKey.sign(signed.bodyBytes);
+		signed.sigMap ={
+			sigPair: [walletKey.publicKey._toProtobufSignature(signature)]
+		}
+
+		const transferSignedTransactionBytes =  SignedTransaction.encode(signed).finish();
+		hederaTx._paymentTransactions.push({
+			signedTransactionBytes: transferSignedTransactionBytes
+		});
+		const response = await hederaTx.execute(this.provider.getHederaClient());
+		// TODO: this may not be the best thing to return but it should work for testing
+		return hexlify(response.asBytes());
```

Comment:
We must parse situations in which the contract call reverts for some reason. See the ethers implementation for reference. In the `provider.call` method if the response code is > 400 exception is thrown and the exception is handled through this:
https://github.com/ethers-io/ethers.js/blob/0578a88efa4f5f65df90a5ac8b63ffdfb5e44d9a/packages/providers/lib/json-rpc-provider.js#L68

We need to have similar experience to failing queries or reverting TX